### PR TITLE
Upgrade deps and point main file to compiled/index.js

### PR DESCRIPTION
**Pull request by hans-lizihan**

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,8 +1,14 @@
 {
-  "plugins": ["transform-object-rest-spread"],
   "presets": [
     [
-      "env"
+      "@babel/preset-env",
+      {
+        "targets": {
+          "node": 14
```
Comment (tdumitrescu) [inline .babelrc:7]: How come Node? Generally we want a set of browser targets, for real browser testing. Typically Node is ahead enough anyway that the UI tests will run fine in the Node env if they're transpiled to support the browser runtimes.

Comment (hans-lizihan) [reply .babelrc:7]: oh yeah nice catch!

I updated to use `targets: "default"` which translates to `> 0.5%, last 2 versions`.

I think the main goal here is remove polyfill related code, our test env in node / browser could all support native async await without regenerator runtime

With that said, the babel only does one thing: transform esm to commonjs.

File: package.json [hunk 2]
```diff
@@ -32,19 +32,19 @@
   },
   "homepage": "https://github.com/mixpanel/domsuite#readme",
   "dependencies": {
+    "@babel/cli": "^7.17.6",
+    "@babel/preset-env": "^7.16.11",
+    "core-js": "^3.21.1",
```
Comment (tdumitrescu) [inline package.json:37]: The babel deps should go back in devDependencies, while core-js stays here. The distinction is whether the code is included/executed at runtime. The babel deps are just for the build phase.

Also pin down the exact version of everything, no ^

Comment (hans-lizihan) [reply package.json:35]: whops, totally messed up with dependencies and devDependencies. Just fixed~

File: package.json [hunk 3]
```diff
@@ -32,19 +32,19 @@
   },
   "homepage": "https://github.com/mixpanel/domsuite#readme",
   "dependencies": {
+    "core-js": "^3.21.1",
```
Comment (tdumitrescu) [inline package.json:35]:
```suggestion
    "core-js": "3.21.1",
```

**Review by tdumitrescu — APPROVED** [review]
Thanks for bearing with me, just the one last change left

**Review by tdumitrescu — COMMENTED** [review]
Oh sorry, one more thing - it looks like this reverted package-lock.json to an older lockfileVersion - can you upgrade to latest npm and rerun `npm install`?

**Comment by tdumitrescu** [comment]
Thanks, published in v0.5.0

**Pull request merged**